### PR TITLE
OAK-11297: remove usage of Guava Joiner

### PR DESCRIPTION
**Pull request by reschke**

### CODE REVIEW

**Review by rishabhdaim — APPROVED** [review]

File: oak-run/src/main/java/org/apache/jackrabbit/oak/index/IndexCommand.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
  */
 package org.apache.jackrabbit.oak.index;
 
-import org.apache.jackrabbit.guava.common.base.Joiner;
+import com.beust.jcommander.Strings;
```
Comment (mbaedke) [inline oak-run/src/main/java/org/apache/jackrabbit/oak/index/IndexCommand.java]: Really?

File: oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCommand.java [hunk 2]
```diff
@@ -212,12 +211,11 @@ private void execute(NodeStoreFixture fixture, DataStoreOptions dataStoreOpts, O
 
                                 while (idIter.hasNext()) {
                                     String id = idIter.next();
-                                    final Joiner delimJoiner = Joiner.on(DELIM).skipNulls();
                                     // If --verbose is present, convert blob ID to a backend friendly format and
                                     // concat the path that has the ref. Otherwise simply add the ID to the o/p file
                                     // as it is.
                                     String line = dataStoreOpts.isVerbose() ?
-                                        VerboseIdLogger.encodeId(delimJoiner.join(id, escapeLineBreak(nodeId)),
+                                        VerboseIdLogger.encodeId(String.join(DELIM, id, escapeLineBreak(nodeId)),
```
Comment (mbaedke) [inline oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCommand.java]: Isn't it theoretically possible that nodeId is null?

**Review by mbaedke — CHANGES_REQUESTED** [review]
Please remove that com.beust.jcommander.Strings thing :)

File: oak-run/src/main/java/org/apache/jackrabbit/oak/index/IndexCommand.java (hunk 1 above)
Comment (reschke) [reply oak-run/src/main/java/org/apache/jackrabbit/oak/index/IndexCommand.java]: Nope.

File: oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCommand.java (hunk 2 above)
Comment (reschke) [reply oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCommand.java]: it might

**Pull request merged**